### ISSUE TITLE
creators search: "Something went wrong" error when entering a few letters in search

### ISSUE THREAD

**Issue by tmorrell** [inveniosoftware/invenio-app-rdm#1849]
Package version (if known): 9.1

## Describe the bug
I observe "something went wrong" errors in the creator search bar when entering a few letters on our instance. The autocomplete works as expected when I complete more of the name (see screenshots). We have a small (~300 names) local vocabulary loaded. I do not observe this behavior on the InvenioRDM sandbox with the entire ORCID vocabulary loaded.

## Steps to Reproduce
1. Go to creators section on deposit form
2. Enter one or two letters in search
3. See error

## Expected behavior
I would not expect to see an error. I'm also not sure the best way to debug this, as I don't see obvious console or log error messages.

## Screenshots (if applicable)

<img width="1106" alt="Screen Shot 2022-08-22 at 1 50 04 PM" src="https://user-images.githubusercontent.com/10798874/186016641-9a0ab0c7-057e-438c-9aa8-b234ae04fdfc.png">
<img width="1106" alt="Screen Shot 2022-08-22 at 1 50 14 PM" src="https://user-images.githubusercontent.com/10798874/186016644-8382cfcd-f221-4a80-b889-c35407db9b8a.png">

**Comment by fenekku** [comment]
We've encountered this for funders as well, so the problem is probably more generalized.

**Comment by tmorrell** [comment]
I've spent some time debugging this, and one issue is when there are names without affiliations. This doesn't happen in the demo site because the test data is cleaned, but it will happen with live ORCID data.

To get the error to show up I had to modify https://github.com/caltechlibrary/react-invenio-forms/blob/176c6367a31fb37d6ae60cf65daf396f0a4acdcd/src/lib/forms/RemoteSelectField.js#L96 with a `console.log(e)`

Then a possible fix is to modify https://github.com/inveniosoftware/react-invenio-deposit/blob/040b698967e7fee14ac7fe466e49f883efbe8c47/src/lib/components/Creatibutors/CreatibutorsModal.js#L234

with something like

```
let idx = 0;
        for (const affiliation of creatibutor.affiliations)
                if (typeof affiliation !== 'undefined'){
                        affNames += affiliation.name;
                        if (idx < creatibutor.affiliations.length - 1) {
                            affNames += ", ";
                                  }
                                idx++;
        };
```

But I haven't been able to get a test instance working with both modified packages loaded so I can't effectively test and see error messages at the same time. I'm not sure if there is a better way to get the errors out of react-invenio-forms without modifying the package. I'm leaving this for someone with more js experience for now.
